WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeActivationBottomSheetFragment.kt

```diff
@@ -29,14 +39,30 @@ class ThemeActivationBottomSheetFragment : BottomSheetDialogFragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        // Ensures that bottom sheet always open in expanded state even in landscape mode
-        val behavior = from(requireView().parent as View)
-        behavior.state = STATE_EXPANDED
+        // Ensures that bottom sheet always opens in expanded state even in landscape mode
+        from(requireView().parent as View).state = STATE_EXPANDED
 
         with(ThemeActivationBottomSheetBinding.bind(view)) {
+            viewModel = ViewModelProvider(requireActivity(), viewModelFactory).get(ThemesViewModel::class.java)
+            viewModel.themeToActivate.observe(this@ThemeActivationBottomSheetFragment) {
+                themeNameTv.text = it.name
+                infoTv.text = requireActivity().getString(R.string.theme_bottom_sheet_info, it.name)
+                useThemeLayoutMainTextTv.text =
+                        requireActivity().getString(R.string.theme_bottom_sheet_use_theme_layout_main_text, it.name)
+            }
+
             closeButton.setOnClickListener { dismiss() }
-            useThemeLayoutLl.setOnClickListener { toggleSelection(useThemeCheckIv, keepCurrentCheckIv) }
-            keepCurrentLayoutLl.setOnClickListener { toggleSelection(keepCurrentCheckIv, useThemeCheckIv) }
+
+            useThemeLayoutLl.setOnClickListener {
```

Comment:
This approach works as expected but we can remove any logic from the Fragment.  I think it might work better if instead we moved all the listener code in the ViewModel like we do with `onActivateMenuItemClicked`. We can then observe the state from the Fragment to make any view related changes.

Reply:
Thank you for tackling this 👍